mmit/core/piecewise_function.cpp

```diff
@@ -19,161 +19,155 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
 #include "piecewise_function.h"
 
+#define TOL 1e-7
```

Comment:
I thought about this a lot and I am convinced that we cannot avoid using a tolerance. The reason is that we will always have to compare floats and this will always be imprecise. For example, when adding to the map, the positions of some breakpoints needs to be compared. The random tests work because we never generate interval bounds at more than 2 decimals.  If we changed the decimals parameter to 9, I'm sure we would get errors. 

I suggest that we round everything to a fixed number of decimals (e.g. 6). Any point that is added should be rounded prior to being inserted into the map. 6 decimals is very reasonable and should not lead to loss of information.

Does this makes sense @tdhock?